erpnext/patches/v16_0/backfill_utm_source_from_legacy_source.py

```
@@ -0,0 +1,78 @@
+		_backfill_one(doctype)
+
+
+def _backfill_one(doctype: str) -> None:
+	table = f"tab{doctype}"
+
+	to_fill = frappe.db.sql(
+		f"""
+		SELECT COUNT(*) FROM `{table}`
+		WHERE (utm_source IS NULL OR utm_source = '')
+		  AND `source` IS NOT NULL AND `source` != ''
+		"""
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)